nestor_api/lib/workflow.py

```diff
@@ -0,0 +1,49 @@
+"""Workflow library"""
+
+from typing import Optional
+
+import nestor_api.lib.config as config
+import nestor_api.lib.git as git
+
+
+def get_apps_to_move_forward(next_step: str) -> dict:
+    """Get a dict of every apps keyed by their name and whose
+    value indicates if the app should progress to next step
+    or not."""
+    # Copy the configuration to ensure strict isolation
+    config_dir = config.create_temporary_config_copy()
+    config.change_environment(next_step, config_dir)
+    project_config = config.get_project_config()
+    previous_step = get_previous_step(project_config, next_step)
+    apps = config.list_apps_config()
+
+    ready_to_progress_app: dict = {}
+
+    for (app_name, app_config) in apps.items():
+        repository_url = app_config["git"]["origin"]
+        app_dir = git.create_working_repository(app_name, repository_url)
+        ready_to_progress_app[app_name] = should_app_progress(app_dir, previous_step, next_step)
```

Comment:
It may be possible to return `app_config` and `app_dir` along with the status because we will probably need them later on. What do you think ?
Or if we don't want to return them, maybe clean `app_dir` to free some disk space.

Reply:
Remove will be enough at the moment, let's modify it only if needed in the future :)